lib/tapioca/compilers/dsl/active_record_enum.rb

```diff
@@ -0,0 +1,112 @@
+# typed: true
+# frozen_string_literal: true
+
+require "parlour"
+
+begin
+  require "active_record"
+  require "action_controller"
+rescue LoadError
+  # means ActiveRecord::Enum is not installed,
+  # so let's not even define the generator.
+  return
+end
+
+module Tapioca
+  module Compilers
+    module Dsl
+      # `RbiGenerator:: ActiveRecordEnum` decorates RBI files for subclasses of `ActiveRecord::Base` which declare
+      # `enum` fields (see https://api.rubyonrails.org/classes/ActiveRecord/Enum.html).
+      #
+      # For example, with the following `ActiveRecord::Base` subclass:
+      #
+      # ~~~rb
+      # class Post < ApplicationRecord
+      #   enum title_type: %i(book all web), _suffix: :title
+      # end
+      # ~~~
+      #
+      # this generator will produce the RBI file `post.rbi` with the following content:
+      #
+      # ~~~rbi
+      # # post.rbi
+      # # typed: true
+      # class Post
+      #   sig { void }
+      #   def all_title!; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def all_title?; end
+      #
+      #   sig { returns(T::Hash[T.any(String, Symbol), Integer]) }
+      #   def self.title_types; end
+      #
+      #   sig { void }
+      #   def book_title!; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def book_title?; end
+      #
+      #   sig { void }
+      #   def web_title!; end
+      #
+      #   sig { returns(T::Boolean) }
+      #   def web_title?; end
+      # end
+      # ~~~
+      class ActiveRecordEnum < Base
+        extend T::Sig
+
+        sig { override.params(root: Parlour::RbiGenerator::Namespace, constant: T.class_of(::ActiveRecord::Base)).void }
+        def decorate(root, constant)
+          return if constant.defined_enums.empty?
+
+          module_name = "#{constant}::EnumMethodsModule"
+          root.create_module(module_name) do |mod|
+            generate_instance_methods(constant, mod)
+          end
+
+          root.path(constant) do |k|
+            k.create_include(module_name)
+
+            constant.defined_enums.each do |name, enum_map|
+              type = type_for_enum(enum_map)
+              create_method(k, name.pluralize, class_method: true, return_type: type)
+            end
+          end
+        end
+
+        sig { override.returns(T::Enumerable[Module]) }
+        def gather_constants
+          ::ActiveRecord::Base.descendants.reject(&:abstract_class?)
+        end
+
+        private
+
+        sig { params(enum_map: T::Hash[T.untyped, T.untyped]).returns(String) }
+        def type_for_enum(enum_map)
+          value_type = enum_map.values.map { |v| v.class.name }.uniq
```

Comment:
I am not seeing this code-path being exercised properly. In order to test this, we need to supply enums like:
```ruby
enum foo: { a: 10, b: "text", c: true, d: [1,2,3] }
```
which should be generating a return type like `T::Hash[T.any(String, Symbol), T.any(Integer, String, TrueType, Array)]` or something like it.

Can we please add a test case for something similar?

Reply:
Also, it would be great if the same test had two enums on a single AR class, one like in my previous comment and another one with all the value types the same like:
```ruby
enum bar: { a: "a", b: "b", c: "c", d: "d" }
```
which should return `T::Hash[T.any(String, Symbol), String]`

(Of course, all of this with more meaningful names please 😄 )